src/StdString.h

```diff
@@ -1,673 +1,236 @@
-	}
-
-	// Array-indexing operators.  Required because we defined an implicit cast
-	// to operator const CT* (Thanks to Julian Selman for pointing this out)
-	CT& operator[](int nIdx)
-	{
-		return MYBASE::operator[](static_cast<MYSIZE>(nIdx));
-	}
-
-	const CT& operator[](int nIdx) const
-	{
-		return MYBASE::operator[](static_cast<MYSIZE>(nIdx));
-	}
-
-	CT& operator[](unsigned int nIdx)
-	{
-		return MYBASE::operator[](static_cast<MYSIZE>(nIdx));
-	}
-
-	const CT& operator[](unsigned int nIdx) const
-	{
-		return MYBASE::operator[](static_cast<MYSIZE>(nIdx));
-	}
-
-	CT& operator[](long unsigned int nIdx){
-	  return MYBASE::operator[](static_cast<MYSIZE>(nIdx));
-	}
-
-	const CT& operator[](long unsigned int nIdx) const {
-	  return MYBASE::operator[](static_cast<MYSIZE>(nIdx));
-	}
-
-	CT& operator[](long long unsigned int nIdx){
-	  return MYBASE::operator[](static_cast<MYSIZE>(nIdx));
-	}
-
-	const CT& operator[](long long unsigned int nIdx) const {
-	  return MYBASE::operator[](static_cast<MYSIZE>(nIdx));
-	}
-#ifndef SS_NO_IMPLICIT_CASTS
-	operator const CT*() const
-	{
-		return this->c_str();
-	}
-#endif
-};
-
-
-// =============================================================================
-//						END OF CStdStr INLINE FUNCTION DEFINITIONS
-// =============================================================================
-
-//	Now typedef our class names based upon this humongous template
-/** @brief Typedef the class names based on the template */
-typedef CStdStr<char>		CStdStringA;	// a better std::string
-#define CStdString				CStdStringA
-
-
-// -----------------------------------------------------------------------------
-// FUNCTIONAL COMPARATORS:
-// REMARKS:
-//		These structs give us functional classes (which may be used in
-//		Standard C++ Library collections and algorithms) that perform
-//		case-insensitive comparisons of CStdString objects.  This is
-//		useful for maps in which the key may be the proper string but
-//		in the wrong case.
-// -----------------------------------------------------------------------------
-
-#define StdStringLessNoCase		SSLNCA
-#define StdStringEqualsNoCase		SSENCA
-
-struct StdStringLessNoCase
-{
-	inline
-	bool operator()(const CStdStringA& sLeft, const CStdStringA& sRight) const
-	{ return ssicmp(sLeft.c_str(), sRight.c_str()) < 0; }
+typedef RString std::string;
+#else
+class RString {
+    std::string s;
+    static int tolower(char c) {
+        // chars must be converted to unsigned chars before passing to std::tolower
+        return std::tolower(static_cast<unsigned char>(c));
+    }
+    static int toupper(char c) {
+        // chars must be converted to unsigned chars before passing to std::tolower
+        return std::toupper(static_cast<unsigned char>(c));
+    }
+
+  public:
+    static const size_t npos = std::string::npos;
+    using size_type = std::string::size_type;
+
+    RString() {}
+    RString(const char * c) : s(c) {}
```

Comment:
also in general throughout this PR, we put * and & next ot the type (e.g. `const char* c`)

Reply:
I was using clang-format. separate PR to add .clang-format, will use that from now on